config/indexer.go

```diff
@@ -16,6 +16,10 @@ type Indexer struct {
 	// individual multihashes within a Put. A value of 1 means no concurrency,
 	// and zero uses the default.
 	CorePutConcurrency int
+	// FreezeAtPercent is the percent used, of the file system that
+	// ValueStoreDir is on, at which to trigger the indexer to enter frozen
+	// mode. A zero value uses the default. A negative value disables freezing.
+	FreezeAtPercent float64
```

Comment:
Should there also be a grace period after which the publishers assigned to a frozen node are reassigned? 

The idea is to give a window to an admininstrator to potentially expand disk and avoid the cost of reassignment. 

In the case of cid.contact deployment we should almost never need to use the freeze feature considering the current disk growth rate, the alert at 20% utilisation which is monitored at the infra level, and the on call rota with runbook entries to expand disk.

Reply:
That could be something configurable in the assigner service, but it seems unnecessary. Each indexer logs its disk usage at every check
```
Disk usage OK   {"usage": "0.13%", "freezeAt": "90%"}
```
When it is within 10% of reaching the freeze threshold, a warning is logged:
```
Disk usage alert     {"usage": "81.77%", "freezeAt": "90%"}
```
When it is within 2% of reaching the freeze threshold, a critical warning is logged:
```
Disk usage critical  {"usage": "88.45%", "freezeAt": "90%"}
```
The usage checks and log messages become more frequent as usage approaches the freeze point.

Reply:
That makes sense thanks Andrew